test/e2e-go/features/followerNode/syncRestart_test.go

```diff
@@ -0,0 +1,127 @@
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package followerNode
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/daemon/algod/api/client"
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+	"github.com/algorand/go-algorand/test/partitiontest"
+)
+
+func TestSyncRestart(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	// Overview of this test:
+	// Start a two-node network--one in follower mode (follower has 0%, secondary has 100%)
+	// with the nodes having a max account lookback of 2.
+	// Advance the primary node to particular rounds, set the follower's sync round
+	// and then advance the follower node as much as possible.
+	// Restart the network and verify that the sync round hasn't advanced.
+
+	// NOTE: with a max account lookback of MAL, and the follower's sync round at SR:
+	// 		 the follower cannot advance past round SR - 1 + MAL
+
+	var fixture fixtures.RestClientFixture
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodesFollower100SecondMaxAccountLookback2.json"))
+
+	defer fixture.Shutdown()
+
+	// sanity check that the follower has the expected max account lookback of 2:
+	followerCtrl, err := fixture.GetNodeController("Follower")
+	a.NoError(err)
+	cfg, err := config.LoadConfigFromDisk(followerCtrl.GetDataDir())
+	a.NoError(err)
+	a.Equal(uint64(2), cfg.MaxAcctLookback)
+
+	waitTill := func(node string, round uint64) {
+		controller, err := fixture.GetNodeController(node)
+		a.NoError(err)
+		err = fixture.ClientWaitForRoundWithTimeout(fixture.GetAlgodClientForController(controller), round)
+		a.NoError(err)
+	}
+
+	getAlgod := func(node string) client.RestClient {
+		controller, err := fixture.GetNodeController(node)
+		a.NoError(err)
+		algod := fixture.GetAlgodClientForController(controller)
+		return algod
+	}
+
+	getRound := func(node string) uint64 {
+		algod := getAlgod(node)
+		status, err := algod.Status()
+		a.NoError(err)
+		return status.LastRound
+	}
+
+	getSyncRound := func() uint64 {
+		followClient := getAlgod("Follower")
+		rResp, err := followClient.GetSyncRound()
+		a.NoError(err)
+		return rResp.Round
+	}
+
+	a.Equal(uint64(1), getSyncRound())
+
+	waitTill("Primary", 3)
+	// with a max account lookback of 2, and the sync round at 1,
+	// the follower cannot advance past round 2 = 1 - 1 + 2
+	waitTill("Follower", 2)
+	a.LessOrEqual(uint64(3), getRound("Primary"))
```

Comment:
Shouldn't these all be GreaterOrEqual? Nothing prevents the primary from advancing, so this check could be flaky unless it gets flipped around.
```suggestion
	a.GreaterOrEqual(uint64(3), getRound("Primary"))
```

Reply:
Sanity checking the conventions here to ensure that `a.LessOrEqual` $\lrarr \leq$ : 

<img width="512" alt="image" src="https://user-images.githubusercontent.com/291133/230476958-1feabecc-c0d3-4b28-ac14-7d2fae5890fb.png">

and indeed: 
```
--- FAIL: TestSomeNumbers (0.00s)
    /Users/zeph/github/algorand/go-algorand/test/e2e-go/features/followerNode/fixture.go:119: 
        	Error Trace:	/Users/zeph/github/algorand/go-algorand/test/e2e-go/features/followerNode/syncRestart_test.go:137
        	Error:      	"3" is not less than or equal to "2"
        	Test:       	TestSomeNumbers
FAIL
FAIL	github.com/algorand/go-algorand/test/e2e-go/features/followerNode	0.427s
FAIL
```

so `getRound("Primary")` can reach for the starts without affecting the assertion's success.